packages/smart-wallet/src/smartWallet.js

```diff
@@ -993,21 +1006,23 @@ export const prepareSmartWallet = (baggage, shared) => {
             // await so that any errors are caught and handled below
             await watchOfferOutcomes(watcher, seatRef);
           } catch (err) {
-            facets.helper.logWalletError('OFFER ERROR:', err);
+            // This block only runs if the block above fails during one vat incarnation.
+            facets.helper.logWalletError('IMMEDIATE OFFER ERROR:', err);
 
-            // Notify the user
+            // Update status to observers
             if (err.upgradeMessage === 'vat upgraded') {
```

Comment:
This should use the std `isUpgradeDisconnection` predicate, which tests a different condition that is actually the correct condition. @Chris-Hibbert and I looked for all occurrences of `'vat upgraded'` and this seems to be the only non-test-code that tests this condition. However, AFAICT, everything that makes an UpgradeDisconnection makes one that satisfies both conditions, so switching to the correct predicate should not change the behavior of the *current* code.

I just checked again on this PR's u14/8997-error-handling fork, and that is true here as well. So please change to

```suggestion
            if (isUpgradeDisconnection(reason)) {
```

Because the two may become inequivalent in a future system, I would really prefer that this change happens in upgrade14 unless there's a good reason not to.

Reply:
Ok, I just saw https://github.com/Agoric/agoric-sdk/pull/8998#discussion_r1504751870 . But this CI has only started spinning. And the test is actually wrong in a way that could effect future code.

Reply:
Gonna punt on this. Resolving per offline convo